### PR TITLE
:art:  Webhook Block Redesign: block redesign #632

### LINKED ISSUE #632: :art: Webhook Block Redesign: block redesign
## User stories

As a user, I want the "Webhook Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:
image

The files can be found at `libs/features/convs-mgr/stories/blocks/library/webhook-block/`. 

The designs are [here](https://www.figma.com/file/ZGHzSstXOYPMFU1EiqjEHD/Untitled?type=design&node-id=1%3A4&mode=design&t=Hf7DxaUUMQ3BWy57-1)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/webhook-block/`.

### PR DESCRIPTION
**Pull request by levy-web**
# Description

Re designed webhook block

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #632 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Screenshot (optional)

Screenshot from 2023-11-30 06-29-45

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/pipes/filter.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    if (value.length == 0 || filterString == '') return value;
+
+    const variables = []
+    for (const variable of value){
+      if (variable.toLowerCase().includes(filterString.toLowerCase())){
+        variables.push(variable);
+      }
+    }
+    return variables;
+  }
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/pipes/filter.pipe.ts:18]: Good job 👍 but simplify this. No need to create the extra list when we can just return a filtered list of value (rename to variables).

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/webhook-block.jsplumb.ts [hunk 2]
```diff
@@ -21,8 +21,8 @@ export function _WebhookBlockDecoratePlumb(block: WebhookBlock, comp: ComponentR
     // Whether the anchor is target (Other Block -> This Block)
     target: true,
     cssClass:"block_endpoint",
-    endpoint: "Rectangle",
-    anchor: "Left",
+    endpoint: "Dot",
+    anchor: [0, 0.05 , 0, 1],
     maxConnections: -1
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/webhook-block.jsplumb.ts:26]: let's leave this as 'Left' for now

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Good job just simplify the filter pipe and resolve conflicts

**Comment by levy-web** [comment]
@LemmyMwaura , I have merged with develop and done with requested changes.

**Comment by LemmyMwaura** [comment]
@levy-web resolve conflicts

**Comment by levy-web** [comment]
@LemmyMwaura I hve resolved the conflicts

File: libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/components/webhook-block/webhook-block.component.ts [hunk 3]
```diff
@@ -21,19 +27,47 @@ export class WebhookBlockComponent implements OnInit {
...
     this.variables$ = this._variablesStore$$.get();
     this.webhookInputId = `webhook-${this.id}`;
+
+    this.webhookForm = new FormGroup({
+
+      search: new FormControl(''),
+      name: new FormControl(''),
+      value: new FormControl('')
+
+    })
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/components/webhook-block/webhook-block.component.ts:63]: The webhook form already exists, you are supposed to modify its values not create a new one

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/components/webhook-block/webhook-block.component.ts [hunk 4]
```diff
@@ -21,19 +27,47 @@ export class WebhookBlockComponent implements OnInit {
...
+    this.webhookForm.addControl('name', this.name);
+    this.webhookForm.addControl('value', this.name);
+
+  }
+
+  onSubmit(){
+    this.variablesToSave = {
+      name:this.webhookForm.value.name,
+      value: this.webhookForm.value.value
+    }
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/webhook-block/src/lib/components/webhook-block/webhook-block.component.ts:70]: don't add any formControls. get the existing ones and mutate their values with what the user adds.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Pull request merged**